pio-project: add page (#5408)
tldr-pages/tldr: pages/common/pio-project.md

# pio project

> > Tool to manage a PlatformIO project.

Reviewer: ```suggestion
> Tool to manage PlatformIO projects.
```

> More information: <https://docs.platformio.org/en/latest/core/userguide/project/index.html>.

> - Initialize a new PlatformIO based project (defaults to current directory):

Reviewer: ```suggestion
- Initialize a new PlatformIO based project (defaults to current directory if `--project-dir` is omitted):
```

> - Initialize a new PlatformIO based project (defaults to current directory if `--project-dir` is omitted):

Reviewer 2: This should be split into 2 examples. One without `--project-dir`, and another with `--project-dir`.

> - Initialize a new PlatformIO based project (when omitting `--project-dir`, it will default to the current working directory):

Reviewer: ```suggestion
- Initialize a new PlatformIO based project (defaults to current directory):
```

`pio project init --project-dir {{path/to/project_directory}}`

> - Initialize a new PlatformIO based project, specifying a board ID and several project options:

Reviewer 2: This is a pretty complicated example. Can we simplify it at all?

Remember the goal in tldr pages isn't to document _everything_ - just the most common options that you'll need.

> - Initialize a new PlatformIO based project, specifying a board ID and several project options (when omitting `--project-dir`, it will default to the current working directory):

Reviewer: ```suggestion
- Initialize a new PlatformIO based project, specifying a board ID and several project options:
```

`pio project init --project-dir {{path/to/project_directory}} --board {{board_id}} --project-option="{{option}}={{value}}}" --project-option="{{option}}={{value}}"`

> - Print configurations of a specific project:

Reviewer 2: Perhaps this will read better?
```suggestion
- Print the configuration of a specific project:
```

`pio project config --project-dir {{path/to/project_directory}}`

> - Print data of a specific project, intended for IDE extensions/plugins:

Reviewer 2: Would this be useful to a user on the command line / terminal? It strikes me that this might be a bit niche to include here if it's intended for plugins or extensions.

Author: I really should focus more on simplicity and usefulness, I'll quickly edit the examples

> - Print configurations of a specific project (when omitting `--project-dir`, it will default to the current working directory):

Reviewer: ```suggestion
- Print configurations of a specific project:
```

> - Print data of a specific project, intended for IDE extensions/plugins (when omitting `--project-dir`, it will default to the current working directory):

Reviewer: ```suggestion
- Print data of a specific project, intended for IDE extensions/plugins:
```

# pio project

> Tool to manage PlatformIO projects.
> More information: <https://docs.platformio.org/en/latest/core/userguide/project/index.html>.

> - Initialize a new PlatformIO based project:

Reviewer 2: Why the `based` here?
```suggestion
- Initialize a new PlatformIO project:
```

Author: There are tools built upon platformio, which use the same project structure. But I guess, that's not important here.

`pio project init`

> - Initialize a new PlatformIO based project in a specific direcdory:

Reviewer 2: Same here.
```suggestion
- Initialize a new PlatformIO project in a specific direcdory:
```

`pio project init --project-dir {{path/to/project_directory}}`

> - Initialize a new PlatformIO based project, specifying a board ID:

Reviewer 2: and here
```suggestion
- Initialize a new PlatformIO project, specifying a board ID:
```

> `pio project init --board {{board_id}}`

Reviewer 2: Maybe we could give some specific examples here. Perhaps in the form `{{foo|bar|...}}`?

Author: Sure
